LFS-79: Servlet for listing users

zm2he opened the pull request:
Servlet for listing users in standalone module accessed via GET request at `http://localhost:8080/home/users`. Full list accessed with admin authentication. Partial lists accessed with no or other authentication.

Commit:
LFS-79: Servlet for listing users
Servlet for listing users in standalone module

sdumitriu committed:
[misc] Improved codestyle

sdumitriu committed:
LFS-79: Servlet for listing users
Made filtering and paging work, added more info

sdumitriu committed:
LFS-79: Servlet for listing users
Made it work for groups, added javadocs

sdumitriu committed:
LFS-79: Servlet for listing users
Renamed module

sdumitriu committed:
LFS-79: Servlet for listing users
Removed useless provisioning file

sdumitriu committed:
LFS-79: Servlet for listing users
Mention default parameter values in the javadoc

sdumitriu committed:
LFS-79: Servlet for listing users
Bugfix: filters stopped working

sdumitriu committed:
LFS-79: Servlet for listing users
Renamed package

sdumitriu committed:
LFS-79: Servlet for listing users
Refactoring: less code duplication

sdumitriu committed:
LFS-79: Servlet for listing users
Also include an URL for accessing each principal

sdumitriu committed:
LFS-79: Servlet for listing users
Include metadata in the response

sdumitriu committed:
[misc] More documentation

zm2he commented:
In line 157 of the servlet, it appears that a second query is being made to get metadata. This concern might be trivial, but I wonder if using only one query would save time.

From my testing so far, the servlet seems to be functioning well. One note is that the groups do not list their constituent users.

sdumitriu commented:
Getting the total number of matching principals is required for good pagination. I agree that it's not very efficient to run another query for all the matching principals, especially a query that actually returns all the matching principals, but I couldn't find a better way to do it using the `UserManager`. If we find that the performance is bad with many users, we may skip the `UserManager` altogether and run our own count query, but given that we don't expect more than 100 users, it's OK as it is.

Not listing group members is intentional. We do return the number of members, which is enough for the table listing groups. The members can be seen on the group's own page, once we have it.

Commit:
LFS-79: Servlet for listing users
Alternate way to get total matching principals without querying twice.

Commit:
LFS-79: Servlet for Listing Users
Clean up commenting and implement requested changes to simplify writePrincipals method.
